code/hud/hudmessage.cpp

```diff
@@ -742,34 +652,28 @@ int hud_get_scroll_max_pos()
 
 	if (Scrollback_mode == SCROLLBACK_MODE_MSGS_LOG) {
 		int count = 0;
-		line_node *ptr;
 		// number of pixels in excess of what can be displayed
 		int excess = Scroll_max - Hud_mission_log_list_coords[gr_screen.res][3];
 
-		if (EMPTY(&Msg_scrollback_used_list) || !HUD_msg_inited) {
+		if (Msg_scrollback_lines.size() <= 1 || !HUD_msg_inited) {
 			max = 0;
-
 		} else {
-			ptr = GET_FIRST(&Msg_scrollback_used_list);
-			while (ptr != END_OF_LIST(&Msg_scrollback_used_list)) {
-				if (ptr->source != HUD_SOURCE_HIDDEN) {
+
+			for (int i = 0; i < (int)Msg_scrollback_lines.size(); i++) {
```

Comment:
The int-casting can be avoided like so:

```suggestion
			for (size_t i = 0; i < Msg_scrollback_lines.size(); i++) {
```

Reply:
Right. Done.